akka-stream/src/main/scala/akka/stream/stage/Stage.scala

```diff
@@ -306,6 +310,125 @@ abstract class AsyncStage[In, Out, Ext]
 }
 
 /**
+ * INTERNAL API
+ */
+private[stream] object TimerStage {
+  final case class Scheduled(timerKey: Any, timerId: Int, repeating: Boolean) extends DeadLetterSuppression
+
+  sealed trait Queued
+  final case class QueuedSchedule(timerKey: Any, interval: FiniteDuration) extends Queued
+  final case class QueuedScheduleOnce(timerKey: Any, delay: FiniteDuration) extends Queued
+  final case class QueuedCancelTimer(timerKey: Any) extends Queued
+
+  final case class Timer(id: Int, task: Cancellable)
+}
+
+/**
+ * This is a variant of [[DetachedStage]] that can receive timer events.
+ *
+ * When overriding the preStart() and postStop() methods make sure that you call
+ * the inherited methods as the timer support relies on these.
+ */
+abstract class TimerStage[In, Out] extends AsyncStage[In, Out, TimerStage.Scheduled] {
```

Comment:
I think named timers, automatic cancellation, pre start queueing, and all such things that this `TimerStage` provides are very important for the usability. So if we support scheduling in the GraphStage it should provide the same.